addons/account/models/account_account.py

```diff
@@ -432,9 +432,12 @@ def unlink(self):
             raise UserError(_('You cannot perform this action on an account that contains journal items.'))
         #Checking whether the account is set as a property to any Partner or not
         values = ['account.account,%s' % (account_id,) for account_id in self.ids]
-        partner_prop_acc = self.env['ir.property'].search([('value_reference', 'in', values)], limit=1)
+        partner_prop_acc = self.env['ir.property'].sudo().search([('value_reference', 'in', values)], limit=1)
```

Comment:
Shouldn't search be replaced by a private method as well and not be called directly via sudo().

Reply:
@keshrath that's why it in a commit `[TOFIX?]` this is to be discussed. There is no native methods for this kind of things, to see if we add one (not sure, it's not frequent)